molecule/tests/roles/confd/variables/haproxy.tmpl.yml

```diff
@@ -0,0 +1,162 @@
+  run_once: true
+  delegate_to: localhost
+  ansible.builtin.copy:
+    dest: "./tmp/haproxy.cluster_vip.defined.tmpl"
+    content: |
+      bind *:{{ haproxy_listen_port.stats }}
+      bind *:{{ haproxy_listen_port.master }}
+      bind *:{{ haproxy_listen_port.replicas }}
+      bind *:{{ haproxy_listen_port.replicas_sync }}
+      bind *:{{ haproxy_listen_port.replicas_async }}
+
+# 🚀 Execute the main task here
+# This task updates the 'haproxy.tmpl' file
+# replacing lines that start with 'bind' and include specific ports
+# The new lines will either bind to the inventory_hostname or the cluster_vip, depending on the specific port.
+- name: Molecule.tests.roles.confd.variables.haproxy.tmpl | Update haproxy.tmpl (replace "bind" for stats)
+  run_once: true
+  delegate_to: localhost
+  ansible.builtin.lineinfile:
+    path: ./tmp/haproxy.cluster_vip.defined.tmpl
+    regexp: "{{ bind_config_with_vip_item.regexp }}"
+    line: "{{ bind_config_with_vip_item.line }}"
+    backrefs: true
+  loop:
+    - regexp: '^.*bind.*:{{ haproxy_listen_port.stats }}$'
+      line: '    bind {{ inventory_hostname }}:{{ haproxy_listen_port.stats }}'
+    - regexp: '^.*bind.*:{{ haproxy_listen_port.master }}$'
+      line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.master }}'
+    - regexp: '^.*bind.*:{{ haproxy_listen_port.replicas }}$'
+      line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas }}'
+    - regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_sync }}$'
+      line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_sync }}'
+    - regexp: '^.*bind.*:{{ haproxy_listen_port.replicas_async }}$'
+      line: '    bind {{ cluster_vip }}:{{ haproxy_listen_port.replicas_async }}'
+  loop_control:
+    loop_var: bind_config_with_vip_item
+    label: "{{ bind_config_with_vip_item.line }}"
+
+# 🖨️ Debugging the established haproxy.tmpl
+- name: Molecule.tests.roles.confd.variables.haproxy.tmpl | Debug haproxy.tmpl
+  run_once: true
+  delegate_to: localhost
+  ansible.builtin.command:
+    cmd: cat ./tmp/haproxy.cluster_vip.defined.tmpl
+  register: output
+- name: Molecule.tests.roles.confd.variables.haproxy.tmpl | Debug haproxy.tmpl content
+  run_once: true
+  ansible.builtin.debug:
+    var: output.stdout_lines
+
+# ✅ Verifying the correctness of the established haproxy.tmpl
+# If the lines are not replaced correctly, the test fails and an error message is displayed
+- name: Molecule.tests.roles.confd.variables.haproxy.tmpl | Validate updated haproxy.tmpl
+  run_once: true
+  ansible.builtin.assert:
+    that:
+      - "output.stdout_lines[0] == '    bind una.name:8080'"
+      - "output.stdout_lines[1] == '    bind fake.vip.url.com:8081'"
+      - "output.stdout_lines[2] == '    bind fake.vip.url.com:8082'"
+      - "output.stdout_lines[3] == '    bind fake.vip.url.com:8083'"
+      - "output.stdout_lines[4] == '    bind fake.vip.url.com:8084'"
```

Comment:
why don't you use variables (from the task "Set template file test data") here and specify values?

Reply:
Because, precisely, during this testing phase, I establish in a fixed manner what I aim to achieve

This is the concept of TDD (Test Driven Development), where I write a test and ensure that the code passes it

Regardless of the code's structure, we can refactor it as desired, but we must always obtain the same result 😇